File: Códigos/Python/QuickSort.py
def quickSort(vetor):
    metodoAux(vetor, 0, len(vetor) - 1)


def metodoAux(vetor, primeiro, ultimo):
    if primeiro < ultimo:
        meio = particionamento(vetor, primeiro, ultimo)

        metodoAux(vetor, primeiro, meio - 1)
        metodoAux(vetor, meio + 1, ultimo)


def particionamento(vetor, primeiro, ultimo):
    pivor = vetor[primeiro]

    esquerda = primeiro + 1
    direita = ultimo

    pronto = False
    while not pronto:

        while esquerda <= direita and vetor[esquerda] <= pivor:
            esquerda = esquerda + 1

        while vetor[direita] >= pivor and direita >= esquerda:
            direita = direita - 1

        if direita < esquerda:
            pronto = True
        else:
            aux = vetor[esquerda]
            vetor[esquerda] = vetor[direita]
            vetor[direita] = aux

    aux = vetor[primeiro]
    vetor[primeiro] = vetor[direita]
    vetor[direita] = aux

    return direita

File: Códigos/Java/QuickSort.java
public class QuickSort {
    public static void quickSort(int[] vetor) {
        metodoAux(vetor, 0, vetor.length - 1);
    }

    public static void metodoAux(int[] vetor, int primeiro, int ultimo) {
        if (primeiro < ultimo) {
            int meio = particionamento(vetor, primeiro, ultimo);
            metodoAux(vetor, primeiro, meio - 1);
            metodoAux(vetor, meio + 1, ultimo);
        }
    }

    public static int particionamento(int[] vetor, int primeiro, int ultimo) {
        int pivor = vetor[primeiro];
        int esquerda = primeiro + 1;
        int direita = ultimo;
        boolean pronto = false;

        while (!pronto) {
            while (esquerda <= direita && vetor[esquerda] <= pivor) {
                esquerda++;
            }

            while (direita >= esquerda && vetor[direita] >= pivor) {
                direita--;
            }

            if (direita < esquerda) {
                pronto = true;
            }
            else {
                int aux = vetor[esquerda];
                vetor[esquerda] = vetor[direita];
                vetor[direita] = aux;
            }
        } //fim do while

        int aux = vetor[primeiro];
        vetor[primeiro] = vetor[direita];
        vetor[direita] = aux;

        return direita;
    } //fim do método
} //fim da classe
